source/postprocess/visualization/shear_stress.cc

```diff
@@ -67,33 +67,42 @@ namespace aspect
-            SymmetricTensor<2,dim> shear_stress = -2.*eta*deviatoric_strain_rate;
-
-            // Add elastic stresses if existent
+            // Compressive stress is negative by the sign convention
+            // used by the engineering community, and as input and used
+            // internally by ASPECT.
+            // Here, we change the sign of the stress to match the
+            // sign convention used by the geoscience community.
+            SymmetricTensor<2,dim> shear_stress;
+
+            // If elasticity is enabled, the deviatoric stress is stored
+            // in compositional fields, otherwise the deviatoric stress
+            // can be obtained from the viscosity and strain rate.
             if (this->get_parameters().enable_elasticity == true)
               {
-                shear_stress[0][0] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xx")];
-                shear_stress[1][1] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_yy")];
-                shear_stress[0][1] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xy")];
+                shear_stress[0][0] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xx")];
+                shear_stress[1][1] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_yy")];
+                shear_stress[0][1] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xy")];
 
                 if (dim == 3)
                   {
-                    shear_stress[2][2] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_zz")];
-                    shear_stress[0][2] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xz")];
-                    shear_stress[1][2] += in.composition[q][this->introspection().compositional_index_for_name("ve_stress_yz")];
+                    shear_stress[2][2] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_zz")];
+                    shear_stress[0][2] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_xz")];
+                    shear_stress[1][2] -= in.composition[q][this->introspection().compositional_index_for_name("ve_stress_yz")];
                   }
               }
+            else
+              {
+                const SymmetricTensor<2,dim> strain_rate = in.strain_rate[q];
+                const SymmetricTensor<2,dim> deviatoric_strain_rate
+                  = (this->get_material_model().is_compressible()
+                     ?
+                     strain_rate - 1./3 * trace(strain_rate) * unit_symmetric_tensor<dim>()
+                     :
+                     strain_rate);
+
+                const double eta = out.viscosities[q];
+                shear_stress -= 2.*eta*deviatoric_strain_rate;
```

Comment:
To be consistent with the spaces above and also my comment above:

```suggestion
                shear_stress = -2. * eta * deviatoric_strain_rate;
```